### PR TITLE
refactor(x/gamm): remove CreatePool and pool id index boilerplate in gamm (merge)

### PR DESCRIPTION
**Pull request by p0mvn**
Closes: #XXX

## What is the purpose of the change

Currently, we update the pool id index in both `x/gamm` and `x/swaprouter`. Since the goal is to fully migrate the index to `x/swaproter`, we start to remove any references it the old `x/gamm` index.

We still need to keep the state entry in `x/gamm`. However, every method, function or query, utilizing it can be refactored to either swithc to querying `x/swaprouter` index or be removed entirely.

## Brief Changelog

As a result, of sunsetting the index in `x/gamm` , the folllowing changes have been made:
- removed `CreatePool` method from `x/gamm`. Its callers instead use `CreatePool` in `x/swaprouter`.
- removed `GetNextPoolIdAndIncrement` function in `x/gamm` since the updates to the index should not happen anymore
- make pool-incentives keeper take swaprouter keeper instead of gamm for retrieving pool indexes

## Testing and Verifying

This change is already covered by existing tests

## Documentation and Release Note

  - Does this pull request introduce a new feature or user-facing behavior changes? no
  - Is a relevant changelog entry added to the `Unreleased` section in `CHANGELOG.md`? no
  - How is the feature or change documented? not applicable

### CODE REVIEW

File: app/keepers/keepers.go [hunk 1]
```diff
@@ -337,17 +350,6 @@ func (appKeepers *AppKeepers) InitNormalKeepers(
 	)
 	appKeepers.TokenFactoryKeeper = &tokenFactoryKeeper
 
-	appKeepers.SwapRouterKeeper = swaprouter.NewKeeper(
```
Comment (p0mvn) [inline app/keepers/keepers.go:340]: Note: Moved up top to be consistent with CL main

File: x/gamm/keeper/grpc_query.go [hunk 2]
```diff
@@ -109,12 +109,12 @@ func (q Querier) Pools(
 	}, nil
 }
 
-// NumPools returns total number of pools.
+// TODO: mark deprecated and move to swaprouter.
```
Comment (p0mvn) [inline x/gamm/keeper/grpc_query.go:112]: Note: tracked in https://github.com/osmosis-labs/osmosis/issues/3556

File: app/keepers/keepers.go [hunk 3]
```diff
@@ -266,6 +266,17 @@ func (appKeepers *AppKeepers) InitNormalKeepers(
 		appKeepers.GetSubspace(twaptypes.ModuleName),
 		appKeepers.GAMMKeeper)
 
+	appKeepers.SwapRouterKeeper = swaprouter.NewKeeper(
+		appKeepers.keys[swaproutertypes.StoreKey],
+		appKeepers.GetSubspace(swaproutertypes.ModuleName),
+		appKeepers.GAMMKeeper,
+		nil,
```
Comment (czarcas7ic) [inline app/keepers/keepers.go:273]: should we leave a comment here to remember to add CL keeper?

**Review by czarcas7ic — APPROVED** [review]
LGTM!

File: app/keepers/keepers.go (hunk 3 above)
Comment (p0mvn) [inline app/keepers/keepers.go:273]:
```suggestion
		nil,  // TODO: add concentrated liquidity keeper once it is merged
```

**Comment by p0mvn** [comment]
Going to merge with 1 approval once CI passes. This has been QAed on `concentrated-liquidity-main`. Additionally, we discussed this change on the call with ValarDragon

**Pull request merged**